pyodide_build/tests/test_venv.py

```diff
@@ -0,0 +1,225 @@
+        "--no-clear",
+        "--no-vcs-ignore",
+        "--symlinks",
+        # "--copies",
+        # "--always-copy",
+        # "--symlink-app-data",
+        "--no-download",
+        "--never-download",
+        "--download",
+        "--extra-search-dir",
+        "--pip",
+        "--setuptools",
+        "--wheel",
+        "--no-setuptools",
+        "--no-wheel",
+        "--no-periodic-update",
+    ]
+
+    assert set(supported_options) == set(expected_options)
+
+
+@pytest.mark.skipif(True, reason="Requiring full Pyodide environment")
+@pytest.mark.parametrize(
+    "options,check_function",
+    [
+        (
+            [],
+            lambda path: (path / "bin" / "python").exists()
+            and (path / "bin" / "pip").exists(),
+        ),
+        (["--clear"], lambda path: (path / "bin" / "python").exists()),
+        (["--no-vcs-ignore"], lambda path: (path / "bin" / "python").exists()),
+        (
+            ["--no-setuptools"],
+            lambda path: not list(path.glob("**/setuptools-*.dist-info")),
+        ),
+        (["--no-wheel"], lambda path: not list(path.glob("**/wheel-*.dist-info"))),
+    ],
+    ids=["default", "clear", "no-vcs-ignore", "no-setuptools", "no-wheel"],
+)
+def test_integration_venv_creation(temp_venv_path, options, check_function):
+    try:
+        venv.create_pyodide_venv(temp_venv_path, options)
+        assert temp_venv_path.exists()
+        assert (temp_venv_path / "bin" / "python").exists()
+        assert (temp_venv_path / "bin" / "pyodide").exists()
+        assert (temp_venv_path / "pip.conf").exists()
+        assert check_function(temp_venv_path)
+
+    except Exception as e:
+        pytest.fail(f"Failed to create virtual environment: {e}")
+
+
+@pytest.mark.skipif(True, reason="Integration test requiring full Pyodide environment")
+@pytest.mark.parametrize(
+    "package,version",
+    [
+        ("pip", "23.0.1"),
+        ("setuptools", "67.6.0"),
+        ("wheel", "0.40.0"),
+    ],
+)
+def test_integration_package_versions(temp_venv_path, package, version):
+    """Test installing specific package versions."""
+    try:
+        venv.create_pyodide_venv(temp_venv_path, [f"--{package}", version])
+
+        # Check that the package is installed with the specified version
+        dist_info_dirs = list(temp_venv_path.glob(f"**/{package}-{version}*.dist-info"))
+        assert len(dist_info_dirs) > 0, f"{package} {version} not found in the venv"
+
+    except Exception as e:
+        pytest.fail(
+            f"Failed to create virtual environment with {package}={version}: {e}"
+        )
+
+
+@pytest.mark.skipif(True, reason="Integration test requiring full Pyodide environment")
+@pytest.mark.parametrize(
+    "packages",
+    [
+        ["six"],  # simple
+        ["numpy"],  # compiled
+        ["six", "numpy"],  # mixed
+    ],
+)
+def test_pip_install(temp_venv_path, packages):
+    """Test that our monkeypatched pip can install packages into the venv"""
```

Comment:
Instead of checking for our `SYSCONFIGDATA` patches and the like, I thought it would be better to see if the end-user functionality is working, as a working `pip` ensures that those patches have worked.